src/native/libs/System.Security.Cryptography.Native/pal_ssl.h

```diff
@@ -152,10 +159,36 @@ Requests that client sends Post-Handshake Authentication extension in ClientHell
 */
 PALEXPORT void CryptoNative_SslSetPostHandshakeAuth(SSL* ssl, int32_t val);
 
-/*=======
+/*
 Sets session caching. 0 is disabled.
 */
-PALEXPORT void CryptoNative_SslCtxSetCaching(SSL_CTX* ctx, int mode);
+PALEXPORT int CryptoNative_SslCtxSetCaching(SSL_CTX* ctx, int mode,  SslCtxNewSessionCallback newCb, SslCtxRemoveSessionCallback removeCb);
+
+/*
+Returns name associated with given ssl session.
+OpenSSL holds reference to it and it must not be freed.
+*/
+PALEXPORT const char* CryptoNative_SslGetServerName(SSL * ssl);
+
+/*
+This function will attach existing ssl session for possible TLS resume.
+*/
+PALEXPORT int32_t CryptoNative_SslSetSession(SSL* ssl, SSL_SESSION *session);
+
+/*
+ * Frees SSL session.
+ */
+PALEXPORT void CryptoNative_SslSessionFree(SSL_SESSION *session);
+
+/*
+ * Get name associated with given SSL_SESSION.
+ */
+PALEXPORT const char* CryptoNative_SslSessionGetHostname(SSL_SESSION *session);
+
+/*
+ * Associate name with given SSL_SESSION.
+ */
+PALEXPORT int CryptoNative_SslSessionSetHostname(SSL_SESSION *session, const char *hostname);
```

Comment:
There are a couple of whitespace/style issues with the pointer types here.  The rest of the PAL uses type-star-space, instead of type-space-star.

Reply:
updated both. (+some other locations)